### PR TITLE
Adds is.oneOf(value, types) and tests

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ljharb** [comment]
Replaces #22

File: test/index.js [hunk 1]
```diff
@@ -667,3 +667,11 @@ test('is.symbol', function (t) {
...
   t.end();
 });
+
+test('is.oneOf', function (t) {
+  t.ok(is.oneOf('foo', ['object', 'array', 'string']), 'oneOf returns true when a match is found');
+  t.notOk(is.oneOf('foo', ['number', 'fn', 'regexp']), 'oneOf returns false when no match is found');
+  t.throws(function() { is.oneOf('foo') }, /array with at least one type/i, 'throws when called with no types');
+  t.throws(function() { is.oneOf('foo', []) }, /array with at least one type/i, 'throws when called with empty types');
+  t.end();
+});
```
Comment (ljharb) [inline test/index.js]: this file needs a trailing newline

File: index.js [hunk 2]
```diff
@@ -776,4 +776,26 @@ is.symbol = function (value) {
...
+ * @param {Mixed|String[]} types types to test against
+ * @returns {boolean}
+ */
+is.oneOf = function (value, types) {
+  if (!is.array(types) || is.array.empty(types)) {
+    throw new TypeError('second argument must be an array with at least one type');
+  }
+
+  for (var i = 0; i < types.length; i++) {
+    var fn = is[types[i]];
```
Comment (ljharb) [inline index.js]: something was bothering me about this PR, and i think this is the line.

What if someone passes the type "oneOf"? What if they pass the type `toString` or `valueOf` or `hasOwnProperty` or `propertyIsEnumerable`?

I don't think I'm comfortable with dynamically pretending "types" are the same thing as "properties on `is`". What other ideas can we come up with for this?

Comment (bwendt-mylo) [inline index.js]: Maybe the convenience isn't worth the overhead? I've gotten by using `is.array(val) || is.object(val) || ...`

**Pull request closed without merging**